gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherSettings.java

```diff
@@ -412,4 +285,196 @@ public long countBytes(PublishRequest request) {
           return request.getSerializedSize();
         }
       };
+
+  public static class Builder extends ServiceApiSettings.Builder {
+    private final ImmutableList<ApiCallSettings.Builder> methodSettingsBuilders;
+
+    private SimpleCallSettings.Builder<Topic, Topic> createTopicSettings;
+    private BundlingCallSettings.Builder<PublishRequest, PublishResponse> publishSettings;
+    private SimpleCallSettings.Builder<GetTopicRequest, Topic> getTopicSettings;
+    private PageStreamingCallSettings.Builder<ListTopicsRequest, ListTopicsResponse, Topic>
+        listTopicsSettings;
+    private PageStreamingCallSettings.Builder<
+            ListTopicSubscriptionsRequest, ListTopicSubscriptionsResponse, String>
+        listTopicSubscriptionsSettings;
+    private SimpleCallSettings.Builder<DeleteTopicRequest, Empty> deleteTopicSettings;
+
+    private static final ImmutableMap<String, ImmutableSet<Status.Code>> RETRYABLE_CODE_DEFINITIONS;
+
+    static {
+      ImmutableMap.Builder<String, ImmutableSet<Status.Code>> definitions = ImmutableMap.builder();
+      definitions.put(
+          "idempotent",
+          Sets.immutableEnumSet(
+              Lists.<Status.Code>newArrayList(
+                  Status.Code.DEADLINE_EXCEEDED, Status.Code.UNAVAILABLE)));
+      definitions.put("non_idempotent", Sets.immutableEnumSet(Lists.<Status.Code>newArrayList()));
+      RETRYABLE_CODE_DEFINITIONS = definitions.build();
+    }
+
+    private static final ImmutableMap<String, RetrySettings.Builder> RETRY_PARAM_DEFINITIONS;
+
+    static {
+      ImmutableMap.Builder<String, RetrySettings.Builder> definitions = ImmutableMap.builder();
+      RetrySettings.Builder settingsBuilder = null;
+      settingsBuilder =
+          RetrySettings.newBuilder()
```

Comment:
@eschapira Any reason Pub/Sub [SLA](https://cloud.google.com/pubsub/sla) does not specify "Back-off Requirements" like many other [services](https://cloud.google.com/datastore/sla).
Do these look correct to you?